apps/builder/app/builder/features/style-panel/sections/advanced/advanced.tsx

```diff
@@ -103,296 +64,21 @@ const AdvancedStyleSection = (props: {
-        // This helps in preventing layout shifts and provides a better user experience by maintaining a consistent layout.
-        // It also affects the contentvisibilityautostatechange event to be called properly,
-        // with "auto" it will call it with skipped false for all initial elements.
-        // 44px is the height of the property row with 2 lines of text. This value can be adjusted slightly.
-        containIntrinsicSize: "auto 44px",
-      }}
-    >
-      {isVisible ? children : undefined}
-    </div>
-  );
-};
-
-const AdvancedDeclarationLonghand = memo(
-  ({
-    property,
-    onChangeComplete,
-    onReset,
-    valueInputRef,
-    indentation = initialIndentation,
-  }: {
-    property: CssProperty;
-    indentation?: string;
-    onReset?: () => void;
-    onChangeComplete?: ComponentProps<
-      typeof CssValueInputContainer
-    >["onChangeComplete"];
-    valueInputRef?: RefObject<HTMLInputElement>;
-  }) => {
-    return (
-      <Flex
-        css={{ paddingLeft: indentation }}
-        wrap="wrap"
-        align="center"
-        justify="start"
-        {...{ [copyAttribute]: property }}
-      >
-        <AdvancedPropertyLabel property={property} onReset={onReset} />
-        <Text
-          variant="mono"
-          // Improves the visual separation of value from the property.
-          css={{
-            textIndent: "-0.5ch",
-            fontWeight: "bold",
-          }}
-        >
-          :
-        </Text>
-        <AdvancedPropertyValue
-          property={property}
-          onChangeComplete={onChangeComplete}
-          onReset={onReset}
-          inputRef={valueInputRef}
-        />
-      </Flex>
-    );
-  }
-);
-
 export const Section = () => {
-  const [isAdding, setIsAdding] = useState(false);
-  const advancedStyles = useStore($advancedStylesLonghands);
+  const styleMap = useStore($advancedStylesLonghands);
+  const apiRef = useRef<CssEditorApi>();
+  const properties = Array.from(styleMap.keys()) as Array<CssProperty>;
   const selectedInstanceKey = useStore($selectedInstanceKey);
   // Memorizing recent properties by instance id, so that when user switches between instances and comes back
   // they are still in-place
   const [recentPropertiesMap, setRecentPropertiesMap] = useState<
     Map<string, Array<CssProperty>>
   >(new Map());
-  const addPropertyInputRef = useRef<HTMLInputElement>(null);
-  const recentValueInputRef = useRef<HTMLInputElement>(null);
-  const searchInputRef = useRef<HTMLInputElement>(null);
-  const [searchProperties, setSearchProperties] =
-    useState<Array<CssProperty>>();
-  const containerRef = useRef<HTMLDivElement>(null);
-  const [minHeight, setMinHeight] = useState<number>(0);
-
-  const advancedProperties = Array.from(
-    advancedStyles.keys()
-  ) as Array<CssProperty>;
 
   const recentProperties = selectedInstanceKey
     ? (recentPropertiesMap.get(selectedInstanceKey) ?? [])
     : [];
```

Comment:
Why access recent properties by instance key if they are scoped to instance?

Reply:
because we keep that map when instance switches, we don't unmount the panel, so when we come back, we still see the recently added properties for that instance